tests/run/constant_folding.py

```diff
@@ -494,12 +494,14 @@ def cascaded_cmp_with_partial_constants_and_false_end(a, b, c):
     '//PrimaryCmpNode',
     '//PrimaryCmpNode//IntNode',
     '//PrimaryCmpNode//IntNode[@value = "0"]',
-    '//PrimaryCmpNode//IntNode[@value = "4294967296"]',
+    '//PrimaryCmpNode//IntNode[@value = "0x100000000"]',
 )
 @cython.test_fail_if_path_exists(
     '//PrimaryCmpNode//IntBinopNode',
     '//PrimaryCmpNode//IntNode[@value = "1"]',
+    '//PrimaryCmpNode//IntNode[@value = "0x1"]',
     '//PrimaryCmpNode//IntNode[@value = "32"]',
+    '//PrimaryCmpNode//IntNode[@value = "0x20"]',
```

Comment:
Seeing this, what would you think about requiring that the hex string must be shorter than the decimal string in order to replace it? That way, we'd get normal looking decimal strings for anything that users would want to look at, and hex strings for anything that's long enough to merit representing in hex. (This would also merit being a helper function.)